packages/builder/src/components/application/FullPreview.tsx

```diff
@@ -0,0 +1,177 @@
+import { useEffect } from "react";
+import { EyeIcon } from "@heroicons/react/24/solid";
+import { ChainId, renderToHTML } from "common";
+import { Metadata, RoundApplicationQuestion } from "../../types";
+import { RoundApplicationAnswers } from "../../types/roundApplication";
+import { DefaultProjectBanner, DefaultProjectLogo } from "../../assets";
+import Button, { ButtonVariants } from "../base/Button";
+import { AboutProject } from "./AboutProject";
+import { ProjectTitle } from "./ProjectTitle";
+
+export function FullPreview(props: {
```

Comment:
We should state return types, otherwise a function could return something that is not React-renderable data and cause a runtime error. This might sound unlikely, but it's enough to forget a `return` inside a component that conditionally renders based on `if` or `switch` for things to explode (`undefined`is one such invalid data type).

I usually do:

```tsx
export const Foo: React.FC<{ bar: string, baz: number}> = ({ bar, baz}) => {
  // ...
}
```

That takes care of both argument and return type. Not sure how to write the same without `const`, though.

Reply:
Good catch, I actually saw an instance of this in our codebase, where someone forgot to put return in a conditional (maybe they were thinking about Rust-style returns :D ), luckily there it just results in a part of the component not rendering I think. 

I'll resolve.

Reply:
Actually, i just now realized - undefined is a valid type tha React can render. if you look at the ReactNode type: https://reactnative.dev/docs/react-node (applies to react-dom too), undefined and null are just ignored. So I think it's safe to just type the return value as a React.ReactNode to ensure we don't accidentally return a function or something that react can't render. What do you think?

Reply:
Yes, you're right, `undefined` has become ok in React 18 and that's what the project is using.

I think the best would still be to type the function itself as `React.FC<PropType>`, as that will always be valid regardless of React version, but if not possible, adding types manually is still better than nothing.